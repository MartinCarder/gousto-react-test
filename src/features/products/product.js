import React from "react";
import PropTypes from "prop-types";

const Product = ({ title, description, open, toggleOpenProduct, id }) => {
  const productClick = () => toggleOpenProduct(id);
  return (
    <div onClick={productClick}>
      {title}
      {open && <div>{description}</div>}
    </div>
  );
};

Product.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  open: PropTypes.bool.isRequired,
  toggleOpenProduct: PropTypes.func.isRequired
};

export default Product;
