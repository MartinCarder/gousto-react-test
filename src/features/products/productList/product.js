import React from "react";
import PropTypes from "prop-types";

import { Description, Title } from "./product.styled"

const Product = ({ title, description, open, toggleOpenProduct, id }) => {
  const productClick = () => toggleOpenProduct(id);
  return (
    <div onClick={productClick}>
      <Title open={open}>{title}</Title>
      {open && <Description>{description}</Description>}
    </div>
  );
};

Product.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  open: PropTypes.bool.isRequired,
  toggleOpenProduct: PropTypes.func.isRequired
};

export default Product;
