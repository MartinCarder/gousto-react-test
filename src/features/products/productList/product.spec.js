import React from "react";
import { shallow } from "enzyme";

import Product from "./product";
import { Description, Title } from "./product.styled";

describe("Product", () => {
  let wrapper;
  const props = {
    title: "A title",
    description: "A description",
    open: false,
    toggleOpenProduct: jest.fn(),
    id: "1234"
  };

  beforeEach(() => {
    wrapper = shallow(<Product {...props} />);
    jest.resetAllMocks();
  });

  it("Renders and matches snapshoot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("displays correct title", () => {
    expect(wrapper.find(Title).text()).toEqual(props.title);
  });

  it("Don't show description if open false", () => {
    expect(wrapper.find(Description).length).toEqual(0);
  });

  it("Show description if open true", () => {
    wrapper.setProps({
      open: true
    });
    expect(wrapper.find(Description).length).toEqual(1);
  });

  it("Show correct description if open true", () => {
    wrapper.setProps({
      open: true
    });
    expect(wrapper.find(Description).text()).toEqual(props.description);
  });

  it("Clicking on title fires toggleOpenProduct with id", () => {
      wrapper.simulate('click');

      expect(props.toggleOpenProduct.mock.calls.length).toEqual(1);
      expect(props.toggleOpenProduct.mock.calls[0][0]).toEqual(props.id)
  })
});
