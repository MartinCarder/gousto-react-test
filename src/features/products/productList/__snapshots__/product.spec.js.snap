// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Renders and matches snapshoot 1`] = `
<div
  onClick={[Function]}
>
  <styled.div
    open={false}
  >
    A title
  </styled.div>
</div>
`;
